OE-580: New app nav

samanthaandrews opened the pull request.

samanthaandrews commented on cypress/support/commands.ts:
```diff
@@ -82,11 +82,17 @@ Cypress.Commands.add(
     cy.findByRole("textbox", { name: "Access Code input" }).type(username, {
-      log: false
+      log: false,
+      scrollBehavior: false
```
Since the header is now sticky, Cypress tests were failing because they navigate to an element and then the header covers the element. The workaround is to change the `scrollBehavior: false` as you'll see throughout our tests. Annoying.

kristojorg replied:
Oh that is rly annoying haha

samanthaandrews commented on src/components/chakra-ui/theme/foundations/color.ts:
```diff
@@ -43,6 +43,7 @@ const colors: Colors = {
     },
     gray: {
       dark: "#8A8A8A",
+      darkish: "#909090",
```
i realize this naming is pretty terrible lol. any ideas for a better name??

kristojorg replied:
lol hmm "darker" is slightly better buuuut idk

Toxiapo replied:
+1 on darker haha, or maybe ask/update Jen, maybe it will be used on several other places, in that case, we might want a consistent name for it.

kristojorg commented:
<img width="751" alt="image" src="https://user-images.githubusercontent.com/4398775/193331361-26685222-0938-437c-9998-f0839b02c6c1.png">
So I think changing the theme breakpoints might not be a good idea, since the whole marketing site currently depends on those breakpoints. In general, the designs should specify how something responds to the same breakpoints everywhere, though I understand we didn't have designs which accommodated  the "All Books" button. In this case, I think it's probably better to add a breakpoint inline for that specific component, not change them all. We spent significant time with Jen refining how the marketing homepage responds to different sizes, and future designs will use those same breakpoints, so I feel like we shouldn't mess with them

kristojorg commented:
I also am wondering if we should disable the search button once a user initiates a search but before the page has reloaded? Currently it feels like there is a confusing delay after clicking search where nothing happens. Or maybe a loading state?

kristojorg reviewed (changes requested):
whoops meant to request those few changes before approving

Toxiapo commented on src/components/chakra-ui/AuthButton.tsx:
```diff
@@ -0,0 +1,113 @@
+export type AuthButtonProps = {
+  colorScheme?: string;
+  as?: React.ReactElement;
+} & ThemingProps &
```
I don't think you need 'ThemingProps", ChakraProps should be enough.

Toxiapo commented on src/components/chakra-ui/icons/Educators.tsx:
```diff
@@ -1,8 +1,8 @@
 
-const EducatorsIcon = (props: React.SVGProps<SVGSVGElement> & SxProp) => (
-  <svg
+const Educators = (props: React.ComponentProps<typeof Icon>) => (
+  <Icon
     width="24"
     height="24"
```
Since the svg is being replaced with an `Icon`, these values won't work anymore, you might want to update them to something like `width="24px" / height="24px"`

Toxiapo reviewed (changes requested):
Codes look good!

I agree with Kristo regarding the breakpoint, if it's just a one-off breakpoint, we should just inline it.

Toxiapo commented:
Header UI:
Can we add a little margin to the Search component? It's touching the search bar If you hover over "All Books,"
![Screen Shot 2022-10-03 at 2 53 40 PM](https://user-images.githubusercontent.com/14434233/193657007-091eba3e-d999-4272-b97d-68458d340a3c.png)

samanthaandrews commented on cypress/tests/errors/useSwr.ts:
```diff
@@ -81,7 +81,7 @@ describe("Mock endpoint that catches 401, 404, and 502 errors in the ErrorBounda
     cy.findByRole("button", { name: "Borrow this book" })
       .should("exist")
-      .click();
+      .click({ scrollBehavior: false });
```
Good idea, I made it the default

samanthaandrews commented on src/components/chakra-ui/Carousel/Top10Badge.tsx:
```diff
@@ -66,7 +66,6 @@ const Top10Large = (props: SVGProps<SVGSVGElement> & BadgeProps) => {
       </g>
       <defs>
         <filter
-          id="a"
```
@Toxiapo this was the issue with the top10 badges not displaying. I'm not sure how these ids got removed from the chakra svgs?? It must have been a prettier or eslint thing?

kristojorg replied:
How strange! Nice catch

kristojorg commented:
| If you are on the Home page(web app home), the "All Books" button should be on hover state.

Unfortunately, this will be somewhat difficult currently because the actual URL for different home pages (high school, lower, middle, etc), whereas the all books redirects to "/app" which in turn redirects to "/app/collection/:collectionUrl" after determining your root collection

samanthaandrews commented:
Ok I've addressed the comments from @kristojorg, @Toxiapo, and @jotten.

- The MarketingHeader is now using custom breakpoints so I didn't change the original breakpoints we were using.
- I created a ticket to work on storing a user's root catalog url so that we can use that value to display the correct selected state on the "All books" header link: https://jira.nypl.org/browse/OE-674
- I will chat with Jen about improving the search loading state and create tickets accordingly.

Toxiapo reviewed (approved):
LGTM!

kristojorg reviewed (approved):
Looks good!

kristojorg reviewed (approved):
Yeehaw!
